### PR TITLE
feat: merging of locale files with the same name for `@intlify/unplugin-vue-i18n/messages`

### PR DESCRIPTION
**Pull request by DanielleHuisman**
Currently, `unplugin-vue-i18n` doesn't merge keys for locale files with the same name. However, this would be a useful feature to allow splitting up locales over multiple files. For example:
```
src/locales/user/en.yaml
src/locales/user/nl.yaml
src/locales/group/en.yaml
src/locales/group/nl.yaml
```

```
// Should include locales from all subdirectories by locale
import messages from '@intlify/unplugin-vue-i18n/messages'
```

This issue was also brought up for the deprecated `vite-plugin-vue-i18n` in #3.

This PR implements the merging behaviour. I chose to use `lodash.merge` to ensure the deep merging is production-ready code.

To test the new behaviour, the Vite locales were split up into two files. Without the changes to `unplugin-vue-i18n/src/index.ts`, the tests fail, but with the changes they pass again.

### CODE REVIEW

File: packages/unplugin-vue-i18n/src/index.ts [hunk 1]
```diff
@@ -703,9 +703,11 @@ async function generateBundleResources(
     }
   }
 
-  return `export default {
-  ${codes.join(`,\n`)}
-}`
+  return `import merge from 'lodash.merge'
```
Comment (kazupon) [inline packages/unplugin-vue-i18n/src/index.ts:706]: this code will add `lodash.merge' as dependecy. It means that the code of `lodash.merge'` will be bundled.
As unplulgin-vue-i18n, we must avoid adding dependencies and affecting the code size of the application.
If merge is to be supported, it must have its own merge handling.

**Review by kazupon — CHANGES_REQUESTED** [review]
sorry for my late reply.
I've just reviewed!
please check it!

File: packages/unplugin-vue-i18n/src/index.ts (hunk 1 above)
Comment (DanielleHuisman) [reply packages/unplugin-vue-i18n/src/index.ts]: I replaced `lodash.merge` with a deep merge solution from https://stackoverflow.com/a/34749873.

**Comment by kazupon** [comment]
@DanielleHuisman 
Thanks!

One more request, this feature is for `@intlify/unplugin-vue-i18n/messages` only.
so, we need to update docs (`README.md`)

**Comment by DanielleHuisman** [comment]
> One more request, this feature is for `@intlify/unplugin-vue-i18n/messages` only. so, we need to update docs (`README.md`)

I updated `README.md` with a minimal explanation of the merge feature.

**Review by kazupon — APPROVED** [review]
well done!
Thank you so much!

**Pull request merged**